### ISSUE TITLE
panic on insert command when the specified engine doesn't exist

### ISSUE THREAD

**Issue by yaziine** [genjidb/genji#331]
### What version of Genji are you using?

<pre>
$ genji version
Genji v0.10.0
Genji CLI (devel)
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

Note the wrong engine
```
genji insert -engine bar --db my.db -table foo '{"a": 1}' '{"a": 2}'
```

### What did you expect to see?

Nothing. Only the command being successfully executed.

### What did you see instead?
```
panic: runtime error: invalid memory address or nil pointer dereference
```

<details>
<summary>Stack trace</summary>

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x43980dc]

goroutine 1 [running]:
github.com/genjidb/genji/database.New(0x46e24a0, 0xc00002a088, 0x0, 0x0, 0x46de520, 0x49e87f0, 0x0, 0x0, 0x0)
	/Users/yassine/code/github.com/genjidb/genji/database/database.go:48 +0xdc
github.com/genjidb/genji.New(0x46e24a0, 0xc00002a088, 0x0, 0x0, 0xa, 0xc00002ad20, 0xf)
	/Users/yassine/code/github.com/genjidb/genji/new.go:15 +0x68
main.runInsertCommand(0x46e24a0, 0xc00002a088, 0x7ffeefbffaae, 0x4, 0x7ffeefbffab8, 0x5, 0x7ffeefbffac5, 0x3, 0xc000105d00, 0xc00000fca0, ...)
	/Users/yassine/code/github.com/genjidb/genji/cmd/genji/insert.go:140 +0x1f6
main.main.func1(0xc000068b40, 0x5, 0x8)
	/Users/yassine/code/github.com/genjidb/genji/cmd/genji/main.go:87 +0x227
github.com/urfave/cli/v2.(*Command).Run(0xc000114ea0, 0xc0000688c0, 0x0, 0x0)
	/Users/yassine/dev/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x4ed
github.com/urfave/cli/v2.(*App).RunContext(0xc000105a00, 0x46e24a0, 0xc00002a088, 0xc0000200a0, 0xa, 0xa, 0x0, 0x0)
	/Users/yassine/dev/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x81f
github.com/urfave/cli/v2.(*App).Run(...)
	/Users/yassine/dev/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
	/Users/yassine/code/github.com/genjidb/genji/cmd/genji/main.go:185 +0x87d
```

</details>

**Comment by tzzed** [comment]
What do you think about returning an error if the engine doesn't exists?

**Comment by yaziine** [comment]
Yes we just need to check if the given engine is currently supported and return an error otherwise.

**Comment by tzzed** [comment]
@asdine you can close this issue, it solved by #366.
